packages/core/upload/admin/src/components/UploadAssetDialog/AddAssetStep/FromComputerForm.js

```diff
@@ -1,10 +1,130 @@
-import React from 'react';
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React, { useRef } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
 import { Box } from '@strapi/parts/Box';
+import { Row } from '@strapi/parts/Row';
+import { VisuallyHidden } from '@strapi/parts/VisuallyHidden';
+import { H3 } from '@strapi/parts/Text';
+import { ModalFooter } from '@strapi/parts/ModalLayout';
+import { Button } from '@strapi/parts/Button';
+import AddAssetIcon from '@strapi/icons/AddAsset';
+import { useIntl } from 'react-intl';
+import { getTrad } from '../../../utils';
+import { typeFromMime } from '../../../utils/typeFromMime';
+import { AssetSource } from '../../../constants';
+
+const Wrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+`;
+
+const IconWrapper = styled.div`
+  font-size: ${60 / 16}rem;
+
+  svg path {
+    fill: ${({ theme }) => theme.colors.primary600};
+  }
+`;
+
+const MediaBox = styled(Box)`
+  border-style: dashed;
+`;
+
+export const FromComputerForm = ({ onClose, onAddAsset }) => {
+  const { formatMessage } = useIntl();
+  const inputRef = useRef(null);
+
+  const handleClick = e => {
+    e.preventDefault();
+    inputRef.current.click();
+  };
+
+  const handleChange = () => {
+    const files = inputRef.current.files;
+    const assets = [];
+
+    for (let i = 0; i < files.length; i++) {
+      const file = files.item(i);
+
+      assets.push({
+        source: AssetSource.Computer,
+        type: typeFromMime(file.type),
+        url: URL.createObjectURL(file),
+        ext: file.name.split('.').pop(),
+        mime: file.type,
+      });
+    }
+
+    onAddAsset(assets);
```

Comment:
I am being a little nitpicky here but I think the name of this props should be `onAddAssets`

Reply:
Being nitpicky is key 😝 . I'm correcting! Thanks for the catch 😊